protocol/consistencychecks.go

```diff
@@ -245,7 +245,11 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 	case RegistrationType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		if df.AP.ProofType() == m.ProofOfAbsence {
-			if err := cc.verifyReturnedPromise(df, key); err != nil {
+			tbKey := key
+			if msg.Error == ReqNameExisted {
+				tbKey = df.TB.Value
+			}
```

Comment:
It totally makes sense. I just have few more questions, though.
>  Another more common scenario might be, a user registers normally but then loses their device or deletes their local profile or sets up a new device, or in some other way legitimately generates a new key for their account.

Should `KeyChange` be used here instead?
>  They attempt to register and now they'd get ReqNameExisted but ErrBindingsDiffer.

In this case, how does client distinguish its own registration with other client's registration? Also, if it gets `ReqNameExisted` and `ErrBindingsDiffer` because of other user's registration, will calling `GetKey()` and re-verifying with `HandleResponse(new_key)` make sense?